source/D2Common/include/D2Inventory.h

```diff
@@ -145,6 +146,14 @@ struct D2ItemExtraDataStrc
 };
 #pragma pack()
 
+// Helper functions
+// Check if the header signature is correct. Assumes non null ptr.
+inline bool INVENTORY_CheckSignature(D2InventoryStrc* pInventory) { return pInventory->dwSignature == D2C_InventoryHeader; }
+// Check if ptr is non null and if header signature is correct.
+inline D2InventoryStrc* INVENTORY_GetPtrIfValid(D2InventoryStrc* pInventory) { return (pInventory && INVENTORY_CheckSignature(pInventory)) ? pInventory : nullptr; }
```

Comment:
Most likely a matter of preference and it doesn't matter too much, but I'd probably call these `INVENTORY_ValidateSignature` and `INVENTORY_IsValid`

Reply:
`INVENTORY_ValidateSignature` sounds ok to me, I'll make the change. However `INVENTORY_IsValid` would be a bit weird since `INVENTORY_GetPtrIfValid` returns a pointer ?

Reply:
It's never used like this, though. It's always just used as a boolean condition and could actually also be written as

`inline bool INVENTORY_GetPtrIfValid(D2InventoryStrc* pInventory) { return pInventory && INVENTORY_CheckSignature(pInventory); }`

Anyway, it's not too important